packages/core/src/lib/lit-core.ts

```diff
@@ -645,31 +658,78 @@ export class LitCore {
     const serverKeys: Record<string, JsonHandshakeResponse> = {};
 
     let timeoutHandle: ReturnType<typeof setTimeout>;
+    let canRetry = this.config.retryNodeHandshake ?? true;
```

Comment:
Do we want this defaulted to `true`?

Reply:
If we go with timeouting each node at 7.5 seconds and the full handshake at 20, I would say we do (trying almost 3 times per node)
The other option would be to retain the old default of just the handshake at 20s, and nodes individually do not timeout

I'm open for discussion on this, we could even make them mandatory to force the dev into thinking what to do. My numbers were thinking in one of our customers

Reply:
Interesting. I think we should keep the timeout global for all node handshakes at `20s` and allow users to explicitly turn on `retryNodeHandshake` main reason is many users actually use our sdk in CI which can cause a cascade of retry ops to the nodes which might cause per issues due to request buffer back pressure. 


But if we do not feel this is a valid concern then we can leave the default. This is just something ive noticed when we had the older retry logic.

Reply:
Good point, I think however we should be safe enough of DDOSing ourselves if we set a high enough timeout. Kind of targetting the top 5% anomalies of usage
I asked @anshss in slack what numbers he got in the testing he made last days

Reply:
This makes sense to me.

Reply:
For what I've seen with Ansh I feel like 7.5s is still a safe threshold to avoid retying too much
I'm also not a big fan of releasing turned off things that will likely go unnoticed

But on the other side, I think this will change on the near future with naga once we settle just with the threshold and/or convert lit core to an event emitter. So it is a change that will be changed in a not-so-distant future. Thus forcing our consumers handle breaking changes two times in a close timeframe

I now don't have a preference on the default value of this. What do you think?

Reply:
Linking it here for internal:
https://litprotocol.slack.com/archives/C07DNQVC0FR/p1726051959979759?thread_ts=1725035360.246599&cid=C07DNQVC0FR